src/main/java/lotto/domain/LotteryNumbers.java

```diff
@@ -0,0 +1,62 @@
+package lotto.domain;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class LotteryNumbers {
+
+  public static final int LOTTERY_NUMBERS_SIZE = 6;
+
+  private final List<LotteryNumber> lotteryNumbers;
+
+  public LotteryNumbers() {
+    List<LotteryNumber> generated = LotteryNumbersGenerator.generate(LOTTERY_NUMBERS_SIZE);
+    validate(generated);
+    this.lotteryNumbers = new ArrayList<>(generated).stream()
+        .sorted()
+        .collect(Collectors.toList());
+  }
+
+  public LotteryNumbers(List<Integer> numbers) {
+    List<LotteryNumber> generated = numbers.stream()
+        .map(LotteryNumber::new)
+        .sorted()
+        .collect(Collectors.toList());
+    validate(generated);
+    this.lotteryNumbers = generated;
+  }
+
+  public List<LotteryNumber> toList() {
+    return lotteryNumbers;
+  }
+
+  public WinningLottery rank(LotteryNumbers winning) {
+    if (winning.size() != lotteryNumbers.size()) {
+      return WinningLottery.of(-1);
+    }
+
+    int matchedCount = (int) lotteryNumbers.stream()
+        .filter(winning.lotteryNumbers::contains)
```

Comment:
객체의 필드에 접근해서 연산을 수행하는게아닌 객체에 직접 메세지를 보내 응답을 받도록 수행하는게 좋습니다.
```suggestion
        .filter(winning::contains)
```
이런식으로 말이죠 ㅎ